src/performance/metrics/PerformanceUtil.cpp

```diff
@@ -0,0 +1,1369 @@
+        std::string requestResourceAllocation = "false";
+
+        std::string ip = place.at(0).second;
+        std::string user = place.at(1).second;
+        std::string serverPort = place.at(2).second;
+        std::string isMaster = place.at(3).second;
+        std::string isHostReporter = place.at(4).second;
+        std::string hostId = place.at(5).second;
+        std::string placeId = place.at(6).second;
+
+        if (ip.find("localhost") != std::string::npos) {
+            host = "localhost";
+        } else {
+            host = user + "@" + ip;
+        }
+
+        if (isHostReporter.find("true") != std::string::npos) {
+            std::string hostSearch = "select total_cpu_cores,total_memory from host where idhost='"+hostId+"'";
+            std::vector<vector<pair<string, string>>> hostAllocationList = perfDb.runSelect(hostSearch);
+
+            vector<pair<string, string>> firstHostAllocation = hostAllocationList.at(0);
+
+            std::string totalCPUCores = firstHostAllocation.at(0).second;
+            std::string totalMemory = firstHostAllocation.at(1).second;
+
+            if (totalCPUCores.empty() || totalMemory.empty()) {
+                requestResourceAllocation = "true";
+            }
+        }
+
+        if (isMaster.find("true") != std::string::npos) {
+            collectLocalPerformanceData(isHostReporter, requestResourceAllocation,hostId,placeId);
+        } else {
+            collectRemotePerformanceData(host,atoi(serverPort.c_str()),isHostReporter,requestResourceAllocation,hostId,placeId);
+        }
+    }
+    
+    return 0;
+}
+
+std::vector<Place> PerformanceUtil::getHostReporterList() {
+    std::vector<Place> hostReporterList;
+
+    std::string placeLoadQuery = "select ip, user, server_port, is_master, is_host_reporter,host_idhost,idplace from place"
+                                 " where is_host_reporter='true'";
+    std::vector<vector<pair<string, string>>> placeList = perfDb.runSelect(placeLoadQuery);
+    std::vector<vector<pair<string, string>>>::iterator placeListIterator;
+
+    for (placeListIterator = placeList.begin(); placeListIterator != placeList.end(); ++placeListIterator) {
+        vector<pair<string, string>> place = *placeListIterator;
+        Place placeInfo;
+        std::string host;
+        std::string requestResourceAllocation = "false";
+
+        placeInfo.ip = place.at(0).second;
+        placeInfo.user = place.at(1).second;
+        placeInfo.serverPort = place.at(2).second;
+        placeInfo.isMaster = place.at(3).second;
+        placeInfo.isHostReporter = place.at(4).second;
+        placeInfo.hostId = place.at(5).second;
+        placeInfo.placeId = place.at(6).second;
+
+        hostReporterList.push_back(placeInfo);
+    }
+
+    return hostReporterList;
+
+}
+
+int PerformanceUtil::collectSLAResourceConsumption(std::vector<Place> placeList, std::string graphId,
+        std::string masterIP, int elapsedTime) {
+    /*string slaCategoryId;
```

Comment:
Why do we have this section commented?

Reply:
Removed the commented section from the code

Reply:
Ok